src/test/java/edu/harvard/iq/dataverse/api/FilesIT.java

```diff
@@ -1401,6 +1400,90 @@ public void testDataSizeInDataverse() throws InterruptedException {
+        getFileDataResponse.then().assertThat()
+                .body("status", equalTo("ERROR"))
+                .body("message", containsString("is not permitted to perform requested action."))
+                .statusCode(UNAUTHORIZED.getStatusCode());
+
+        // Publish the dataset with 1 file
+        UtilIT.publishDatasetViaNativeApi(datasetId, "major", superUserApiToken);
+
+        // Regular user can see latest version now
+        getFileDataResponse = UtilIT.getFileVersionsList(dataFileId, regularApiToken, DS_VERSION_LATEST);
+        getFileDataResponse.prettyPrint();
+        getFileDataResponse.then().assertThat()
+                .body("status", equalTo("OK"))
+                .body("data[0].datasetVersion", equalTo("1.0"))
+                .body("data[0].fileDifferenceSummary.file", equalTo("Added"))
+                .statusCode(OK.getStatusCode());
+
+        // Add another file to create a new draft
+        pathToFile = "src/main/webapp/resources/images/dataverseproject.png";
+        addResponse = UtilIT.uploadFileViaNative(datasetId.toString(), pathToFile, superUserApiToken);
+        addResponse.getBody().jsonPath().getString("data.files[0].dataFile.id");
+
+        // Regular user can only see the published version
+        getFileDataResponse = UtilIT.getFileVersionsList(dataFileId, regularApiToken, DS_VERSION_LATEST);
+        getFileDataResponse.prettyPrint();
+        getFileDataResponse.then().assertThat()
+                .body("status", equalTo("OK"))
+                .body("data[0].datasetVersion", equalTo("1.0"))
+                .body("data[0].fileDifferenceSummary.file", equalTo("Added"))
+                .statusCode(OK.getStatusCode());
+        getFileDataResponse = UtilIT.getFileVersionsList(dataFileId, regularApiToken, DS_VERSION_DRAFT);
+        getFileDataResponse.prettyPrint();
+        getFileDataResponse.then().assertThat()
+                .body("status", equalTo("ERROR"))
+                .body("message", containsString("is not permitted to perform requested action."))
+                .statusCode(UNAUTHORIZED.getStatusCode());
+
+        // Give permission to view the draft version
+        Response assignRole = UtilIT.grantRoleOnDataverse(dataverseAlias, DataverseRole.CURATOR.toString(),
+                "@" + regularUsername, superUserApiToken);
+        assertEquals(200, assignRole.getStatusCode());
+
+        // Regular user can see the draft and released versions now
+        getFileDataResponse = UtilIT.getFileVersionsList(dataFileId, regularApiToken, DS_VERSION_DRAFT);
+        getFileDataResponse.prettyPrint();
+        getFileDataResponse.then().assertThat()
+                .body("status", equalTo("OK"))
+                .body("data[0].datasetVersion", equalTo("DRAFT"))
+                .body("data[1].datasetVersion", equalTo("1.0"))
+                .body("data[1].fileDifferenceSummary.file", equalTo("Added"))
+                .statusCode(OK.getStatusCode());
+    }
```

Comment:
Should probably test doing a replace file and verifying that looking for either file returns info about both?